build.sbt

```diff
@@ -211,6 +214,67 @@ lazy val testStdLayout = Project("daffodil-test-stdLayout", file("test-stdLayout
   .dependsOn(tdmlProc % "test")
   .settings(commonSettings, nopublish)
 
+/* Workaround: certain reflection (used by JAXB) isn't allowed by default in JDK 17:
+ * https://docs.oracle.com/en/java/javase/17/migrate/migrating-jdk-8-later-jdk-releases.html#GUID-7BB28E4D-99B3-4078-BDC4-FC24180CE82B
+ *
+ * While we can handle this JVM quirk at build time, at runtime we won't know
+ * a user's JVM version. We'll provide documentation and an extension setting
+ * to add these flags to the extension-launched debugger backend.
+ */
+lazy val extraJvmOptions: Seq[String] =
+  if (scala.util.Properties.isJavaAtLeast("17"))
+    Seq(
+      "--add-opens",
+      "java.base/java.lang=ALL-UNNAMED",
+    )
+  else Seq()
+
+lazy val xjcSettings =
+  Seq(
+    libraryDependencies ++= Seq(
+      "com.sun.xml.bind" % "jaxb-impl" % "2.2.11",
+      "javax.activation" % "activation" % "1.1.1",
+      "org.glassfish.jaxb" % "jaxb-xjc" % "2.2.11",
+    ),
+    xjcCommandLine += "-nv",
+    xjcCommandLine += "-p",
+    xjcCommandLine += "org.apache.daffodil.tdml",
+    xjcBindings += "daffodil-tdml-lib/src/main/resources/bindings.xjb",
+    xjcLibs := Seq(
+      "org.glassfish.jaxb" % "jaxb-xjc" % "2.2.11",
+      "com.sun.xml.bind" % "jaxb-impl" % "2.2.11",
+      "javax.activation" % "activation" % "1.1.1",
+    ),
+    xjcJvmOpts ++= extraJvmOptions,
+    Compile / xjc / sources := Seq(
+      // Need to refer to the individual files because we do not want XMLSchema to go through the JAXB compliation
```

Comment:
```suggestion
      // Need to refer to the individual files because we do not want XMLSchema to go through the JAXB compilation
```
speeling

Reply:
I don't think that's the case - We set `Compile / xjc / sources := Seq()` in `commonSettings` and overwrite it with this - these should be the only files that we generate JAXB for. Potentially, any imports we would also grab, but I believe we have all of the imports explicitly defined here anyways.

I can double check this, but I'm almost positive that either setting `Compile / xjc / sources` to contain the folders or adding XMLSchema won't build because XMLSchema prevents the build from passing (If I'm remembering correctly, there are a lot of clashes, and we would need a lot more definitions in the bindings.xjb file). I also don't see any `xs:include` lines for it, so I don't think it's strictly necessary to include this file for the TDML JAXB bindings to be complete.

Reply:
Double checked - it clashes a lot with XMLSchema_for_DFDL.xsd - about 50 additional entries into the bindings.xjb file would be required.

Reply:
Ah I see, re: `xs:include` wonkiness. Yeah I was hoping to use the default `Compile / xjc / sources` (https://github.com/sbt/sbt-xjc/blob/master/src/main/scala/com/github/retronym/sbtxjc/SbtXjcPlugin.scala#L64).

The only downside is that if there are any file-level xsd additions or deletions then the build file needs to be adjusted.

Reply:
Right. Unless there's a way to use the default and then explicitly exclude XMLSchema.xsd but that isn't perfect either... I'd be okay with either way.

Reply:
I think you should only need to generate objects for:

tdml.xsd

The challenge is that some TDML files embed mini-schemas inside using tdml:defineSchema. 

That can hold anything one would find inside a DFDL schema, so.... not sure how to cope with that without dragging all of XMLSchema_for_DFDL.xsd 

These other schema files should not be needed nor used:

dafint.xsd
dafext.xsd
dfdlx.xsd
datatypes.dtd
xml.xsd
XMLSchema.dtd
XMLSchema.xsd

And hopefully not these depending on the tdml:defineSchema issue:

XMLSchema_for_DFDL.xsd
DFDL_part?_*.xsd - 3 files match this (used by XMLSchema_for_DFDL.xsd)

Reply:
I just did a build, removing every xsd file other than tdml.xsd and removing the reference to bindings.xjb. I get a lot of undefined element messages. The XJC plugin seems to require every file whose definitions we import to go through the JAXB process as well.

Reply:
I suggest maybe this is an approach that could work:

- make a variant of tdml.xsd which omits the tdml:defineSchema element. 

- plan to add back some manual implementation for that object. 

So you need to:
Comment out all the includes/imports.

Comment out line 64 (element ref to tns:defineSchema).

Comment out all other uses of "defineSchema" and the definition of it. 

Will your tool to generate code still work then? That *should* eliminate pulling in all that other stuff and all the excess definitions.

Reply:
I ended up having to take out a bit more than just that, but I was able to generate the XSDs without a bindings.xjb file and only having tdml.xsd listed as a source file. Here's the output from `git diff` in a patch file. If you would like me to upload the entire changed document, let me know.

I will go back and attempt to see if this works with our extension, but it looks like it should work for us since we only use a very minimal subset of the TDML schema for now.

[tdml-xsd-update.patch](https://github.com/apache/daffodil/files/12620820/tdml-xsd-update.patch)

Reply:
The content of defineSchema is used for testing schemas emedded in the TDML file itself.

I guess excluding that from jaxb would be fine if the extension only supports external schemas, which is how most real schema projects are used. So maybe that's a fine limitation? Same probably goes for he defineConfig section?

The most important stuff is probably the test cases with inputs and expected outputs.

Is there a plan so we don't have to maintain two separate tdml.xsd files that are 90% the same except for this small patch?

Reply:
We could have tdml-core.xsd and tdml.xsd which includes tdml-core.xsd.

The jaxb could use only the core, and the full tdml.xsd could add in the missing stuff like defineSchema and provide the root tdml:testSuite element.

Reply:
This change looks like it would work for the extension. There was a change related to namespaces that I'm not thrilled about, but that can more than likely be fixed with a couple more code changes...

We are not planning to support internal schemas in the extension. One of the features we are looking to implement is to provide the ability to package the TDML schema and the DFDL schema in a zip archive, which directly conflicts with having an internal schema, so I don't think we'd be looking at that even for the longer term.

We currently don't support configs in the extension at all, as far as I know... Even if we were to add them in, there's no reason for us to not use them externally, as we're already using the schema.